src/tsxWithoutJsxRule.ts

```diff
@@ -0,0 +1,49 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+
+const FAILURE_STRING: string = 'Files with .tsx/.jsx extension should contain JSX Elements. Rename to .ts/.js instead.';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'tsx-without-jsx',
+        type: 'maintainability',
+        description: "All .tsx/.jsx files should contain JSX elements to clarify the file's purpose.",
+        options: null, // tslint:disable-line:no-null-keyword
```

Comment:
That sounds like another rule ban-file-extension or something similar. This rule explicitly will warn against .tsx and .jsx not using jsx features. The opposite of that behavior would be to warn against .ts and .js using jsx features which doesn't compile and hence doesn't require a lint rule. I understand where you are coming from, but it sounds like we are trying to club unrelated functionality into a rule with a specific purpose.

Reply:
The opposite will  be forcing `.tsx/.jsx` extension even when JSX isn't used in file.
Example repo https://github.com/microsoft/TypeScript-React-Starter and discussion https://github.com/microsoft/TypeScript-React-Starter/issues/4

So rule might require string option: `always` will require all files to have `.tsx/.jsx` extension, `never` will match current behavior and may be default.

Haven't found any other TSLint rules that tracks these cases. Also both options do not contradict to the rule name.

Reply:
It still sounds like a simpler rule to ban file extensions would better accomplish this request as well as have additional value. I'm happy to add it, so that the opposing viewpoint is also represented.